--- lib/studies/shrine/expanding_bottom_sheet.dart
+++ lib/studies/shrine/expanding_bottom_sheet.dart
@@ -586,7 +586,7 @@
     final internalSet = Set<int>.from(_internalList);
     final listSet = Set<int>.from(_list.list);

-    final Set<int> difference = internalSet.difference(listSet);
+    final Set<int> difference = internalSet.difference(listSet).union(listSet.difference(internalSet));
     if (difference.isEmpty) {
       return;
     }
